tests/cl-go-client/main.go

```diff
@@ -231,6 +238,78 @@ func swapGivenLargeAmountsBothDirections(igniteClient cosmosclient.Client, poolI
 	log.Println("finished swapping, num swaps done", numSwaps)
 }
 
+func createExternalCLIncentive(igniteClient cosmosclient.Client, poolId uint64, gaugeCoins sdk.Coins, expectedEpochIdentifier string) {
+	var (
+		randAccountNum = rand.Intn(8) + 1
+		accountName    = fmt.Sprintf("%s%d", accountNamePrefix, randAccountNum)
+	)
+
+	epochsQueryClient := epochstypes.NewQueryClient(igniteClient.Context())
+	currentEpochResponse, err := epochsQueryClient.CurrentEpoch(context.Background(), &epochstypes.QueryCurrentEpochRequest{
+		expectedEpochIdentifier,
+	})
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	log.Println("current epoch", currentEpochResponse.CurrentEpoch, "epoch identifier", expectedEpochIdentifier)
+
+	log.Println("querying epoch info. Note that incentives are distributed at the end of epoch. That's why it matters.")
+	epochInfosRespose, err := epochsQueryClient.EpochInfos(context.Background(), &epochstypes.QueryEpochsInfoRequest{})
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	if len(epochInfosRespose.Epochs) > 0 {
+		lastEpochInfo := epochInfosRespose.Epochs[len(epochInfosRespose.Epochs)-1]
+		log.Println("epoch duration", lastEpochInfo, "next epoch start time", lastEpochInfo.StartTime.Add(lastEpochInfo.Duration))
+	} else {
+		log.Println("could not find information about previous epoch. If duration is too long, this test might be infeasible")
+	}
+
+	//.Create gauge
+	runMessageWithRetries(func() error {
+		return createGauge(igniteClient, expectedPoolId, accountName, gaugeCoins)
+	})
+
+	epochAfterGaugeCreation := int64(-1)
+	for {
+		// Wait for 1 epoch to pass
+		currentEpochResponse, err = epochsQueryClient.CurrentEpoch(context.Background(), &epochstypes.QueryCurrentEpochRequest{
+			expectedEpochIdentifier,
+		})
+		if err != nil {
+			log.Fatal(err)
+		}
+		if epochAfterGaugeCreation == -1 {
+			log.Println("current epoch after gauge creation", currentEpochResponse.CurrentEpoch)
+			log.Println("waiting for next epoch...")
+			epochAfterGaugeCreation = currentEpochResponse.CurrentEpoch
+			continue
+		}
+
+		// One epoch after gauge creation has passed
+		if epochAfterGaugeCreation+1 == currentEpochResponse.CurrentEpoch {
+			log.Println("next epoch reached, checking incentive records...")
+			break
+		}
+
+		log.Println("current epoch", currentEpochResponse.CurrentEpoch)
+		time.Sleep(5 * time.Second)
+	}
+
+	clQueryClient := clqueryproto.NewQueryClient(igniteClient.Context())
+
+	incentiveRecordsResponse, err := clQueryClient.IncentiveRecords(context.Background(), &clqueryproto.IncentiveRecordsRequest{
+		PoolId: expectedPoolId,
+	})
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	log.Println("incentive records. If empty, something probably went wrong", incentiveRecordsResponse.IncentiveRecords)
```

Comment:
can we also check gauge Records after we distribute? FilledEpoch should be +1 and distribtued Coins should be appended to the gauge struct

wait also i'm surprised this didn't fail here(https://github.com/osmosis-labs/osmosis/blob/036b94d3e220959e1ecc6050674c06de60f548f6/x/incentives/keeper/distribute.go#L393)

Reply:
also should we consider all the created incentives regardless of poolId <> gaugeId association and distribute tokens to them as well?

Reply:
The purpose of this is to make sure that incentive records are created. This print gets the job done.

I don't think what you're requesting is helpful for this tool. However, it would be a good assertion to make in tests